distro/src/main/java/io/camunda/migrator/app/MigratorApp.java

```diff
@@ -104,23 +105,30 @@ protected static void validateArguments(String[] args) {
       }
     }
 
+    // Check if we have too many flags (not counting entity type parameters)
+    if (flagCount > MAX_FLAGS) {
+      throw new IllegalArgumentException("Error: Too many arguments.");
+    }
+
+    // Validate skipped migration configuration
+    if (hasListSkipped && hasRetrySkipped) {
+      throw new IllegalArgumentException(
+          "Conflicting flags: --list-skipped and --retry-skipped cannot be used together. Please specify only one migration mode.");
```

Comment:
❓ Question just for clarity on terminology: for me I assumed migration mode means history or runtime and the other flags are just adjustments of the behaviour for each mode - do we have any agreed terminlogy here or is any flag also a migrator mode?

Reply:
I will change it to avoid confusion